llvm/tools/sycl-post-link/SpecConstants.cpp

```diff
@@ -489,9 +498,27 @@ PreservedAnalyses SpecConstantsPass::run(Module &M,
           IDs.push_back(NextID);
         }
 
+        Constant *DefaultValue = nullptr;
+        if (F.getName().startswith(SYCL_GET_SCALAR_2020_SPEC_CONST_VAL) ||
+            F.getName().startswith(SYCL_GET_COMPOSITE_2020_SPEC_CONST_VAL)) {
```

Comment:
We can declare `Is2020Intrinsic` variable as on the 556-558 lines, and move it near `IsComposite` definition to avoid duplication:: 
``` 
bool Is2020Intrinsic =
      F.getName().startswith(SYCL_GET_SCALAR_2020_SPEC_CONST_VAL) ||
      F.getName().startswith(SYCL_GET_COMPOSITE_2020_SPEC_CONST_VAL);
...
if (Is2020Intrinsic) { ... }
```

Reply:
Fixed in a5902e417a27d1338cee6487cdb35081f87e0601